sources/packages/web/src/services/http/common/HttpBaseClient.ts

```diff
@@ -62,6 +62,43 @@ export default abstract class HttpBaseClient {
     }
   }
 
+  /**
+   * Http call to download a file as response from API.
+   ** When payload is passed, the file is downloaded on post call
+   ** otherwise it is downloaded as get call.
+   * @param url url of the API.
+   * @param payload payload passed for post call.
+   * @returns axios response object from http response.
+   */
+  protected async downloadFile<T>(
+    url: string,
+    payload?: T,
+  ): Promise<AxiosResponse<any>> {
+    try {
+      const requestConfig: AxiosRequestConfig = {
+        ...this.addAuthHeader(),
+        responseType: "blob",
+      };
+      let response: AxiosResponse<any>;
```

Comment:
The response variable can be removed and the result can be returned directly, which means that the `else` can also be removed and the method complexity can be simplified.